public/javascripts/payments.js

```diff
@@ -315,220 +303,214 @@
+            // Display the Multibanco payment information to the user.
+            const multibanco = source.multibanco;
+            receiverInfo.innerHTML = `
+              <ul>
+                <li>
+                  Amount (Montante):
+                  <strong>${amount}</strong>
+                </li>
+                <li>
+                  Entity (Entidade):
+                  <strong>${multibanco.entity}</strong>
+                </li>
+                <li>
+                  Reference (Referencia):
+                  <strong>${multibanco.reference}</strong>
+                </li>
+              </ul>`;
             break;
           default:
-            // Order is received, pending payment confirmation.
-            break;
+            console.log('Unhandled receiver flow.', source);
         }
+        // Poll the PaymentIntent status.
+        pollPaymentIntentStatus(paymentIntent.client_secret);
         break;
-
-      case 'pending':
-        // Success! Now waiting for payment confirmation. Update the interface to display the confirmation screen.
-        mainElement.classList.remove('processing');
-        // Update the note about receipt and shipping (the payment is not yet confirmed by the bank).
-        confirmationElement.querySelector('.note').innerText =
-          'We’ll send your receipt and ship your items as soon as your payment is confirmed.';
-        mainElement.classList.add('success');
-        break;
-
-      case 'failed':
-        // Payment for the order has failed.
-        mainElement.classList.remove('success');
-        mainElement.classList.remove('processing');
-        mainElement.classList.remove('receiver');
-        mainElement.classList.add('error');
-        break;
-
-      case 'paid':
-        // Success! Payment is confirmed. Update the interface to display the confirmation screen.
-        mainElement.classList.remove('processing');
-        mainElement.classList.remove('receiver');
-        // Update the note about receipt and shipping (the payment has been fully confirmed by the bank).
-        confirmationElement.querySelector('.note').innerText =
-          'We just sent your receipt to your email address, and your items will be on their way shortly.';
-        mainElement.classList.add('success');
+      default:
+        // Customer's PaymentIntent is received, pending payment confirmation.
         break;
     }
   };
 
   /**
    * Monitor the status of a source after a redirect flow.
    *
-   * This means there is a `source` parameter in the URL, and an active order.
-   * When this happens, we'll monitor the status of the order and present real-time
+   * This means there is a `source` parameter in the URL, and an active PaymentIntent.
+   * When this happens, we'll monitor the status of the PaymentIntent and present real-time
    * information to the user.
    */
 
-  const pollOrderStatus = async (
-    orderId,
+  const pollPaymentIntentStatus = async (
```

Comment:
What are we polling this payment intent for? It's being called for sources that don't use PaymentIntents so I'm a little unclear how this works into those source flows (I haven't looked much at how sources interact with PaymentIntents)

Reply:
For source types that aren't yet supported by PI I'm listening to the `source.chargeable` webhook and upon receiving I confirm the PaymentIntent server-side: https://github.com/stripe/stripe-payments-demo/pull/31/files?utf8=%E2%9C%93&diff=split#diff-e096badd84d65c3fbe372dced8268d69R93

Therefore I need to poll the PI on the client-side to know when it's status changed.

Reply:
For source types that aren't yet supported by PI I'm handling the source activation myself (handleSourceActivation()) and then I'm listening to the `source.chargeable` webhook. Upon receiving it, I confirm the PaymentIntent server-side: https://github.com/stripe/stripe-payments-demo/pull/31/files?utf8=%E2%9C%93&diff=split#diff-e096badd84d65c3fbe372dced8268d69R93

Therefore I need to poll the PI on the client-side to know when its status changed.